### ISSUE TITLE
Trying to train models with multiple GPUs on multiple nodes without success.

### ISSUE THREAD

**Issue by RCX112** [NVlabs/stylegan3#117]
I am currently training a model to generate different images of leaves. The first network was trained on a single HPC-Node, which has 4 GPUs.

In the next step I wanted to train a GAN model on 8 GPUs, which means that I would need two compute nodes. I tested it using a slurm script with the srun command but had no success. My guess is that this only works on a single node because the train method isn't using the DistributedDataParallel library from PyTorch and no global parameters for the rank and cuda device. 

I tried modifying the code a little bit using environment variables like SLURM_PROCID or WORLD_SIZE. I also added global and local parameters for the rank and for the GPU count. I also modified the temp_dir variable (used in the torch.distributed.init_process_group method) and changed it to a shared directory. 

I've attached an example of a slurm script where I used 2 nodes, each with 4 GPUs: [slurm_stylegan3_jureca_distributed.txt](https://github.com/NVlabs/stylegan3/files/8025090/slurm_stylegan3_jureca_distributed.txt)

**Comment by RCX112** [comment]
Update: I created a [fork](https://github.com/RCX112/stylegan3) of the repository and tried to run the training on multiple nodes. For that I've edited the [train.py](https://github.com/NVlabs/stylegan3/compare/main...RCX112:main#diff-ed183d67207df065a11e1289f19d34cc2abbc5448dea952683cfe9728c342b95) and [training/training_loop.py](https://github.com/NVlabs/stylegan3/compare/main...RCX112:main#diff-ddf22c2ea2b41ca0f4fb65128ed8b13d1af8f1362f2685be5e4b8d08b7d41d0b) file (and also the [torch_utils/custom_ops.py](https://github.com/NVlabs/stylegan3/compare/main...RCX112:main#diff-994cd89022ad404d5110013114012f88d88b3da931472de195d20026d3fbf791) because I get loading errors for the bias_act_plugin). The training actually runs with no errors when executing a slurm script like [this](https://github.com/RCX112/stylegan3/blob/main/slurm/sbatch_stylegan3_distributed). The resulting images will also get better and better, as expected. But I am not sure if it really runs on multiple nodes.

**Comment by lijain** [comment]
There seems to be a problem. Multilevel multicard cannot be activated

**Comment by SamSilmarilData** [comment]
Hi, did the multinodal training work? i'm trying to use the code that you provided but for some reason the FID_score metric won't go down

**Comment by hugosoli** [comment]
Hi, did somebody around here manage to work this code on miltinode systems. I have 3 nodes with 2GPUs each. Pretty much any code that seems properly created seems to work, I can see that my 6GPUs are working but for some reason 6GPUs works  slower thhan using only one computer. Any ideas
cheers
